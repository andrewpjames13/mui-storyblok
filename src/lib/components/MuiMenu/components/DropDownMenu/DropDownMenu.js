import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import Paper from '@material-ui/core/Paper';
import ArrowDropDown from '@material-ui/icons/ArrowDropDown';
import styles from './DropDownMenu.module.scss';

const DropDownMenu = ({
  children,
  btnTxt,
  color,
  variant,
  size,
  rootClass,
}) => (
  <div className={styles.dropdown}>
    <Button
      className={rootClass}
      color={color}
      variant={variant}
      size={size}
      endIcon={<ArrowDropDown />}
    >
      {btnTxt}
    </Button>
    <Paper
      variant="outlined"
      className={styles.dropdownContent}
    >
      {children}
    </Paper>
  </div>
);

export default DropDownMenu;

DropDownMenu.propTypes = {
  /** button text for dropdown click */
  btnTxt: PropTypes.string,
  /** mui prop: 'default'| 'inherit'| 'primary'| 'secondary' */
  color: PropTypes.string,
  /** mui prop: 'text' |'outlined'| 'contained */
  variant: PropTypes.string,
  /** mui prop: 'small'| 'medium'| 'large' */
  size: PropTypes.string,

  rootClass: PropTypes.string,

  children: PropTypes.node.isRequired,
};

DropDownMenu.defaultProps = {
  rootClass: '',
  variant: 'text',
  btnTxt: '',
  color: 'secondary',
  size: 'medium',
};
