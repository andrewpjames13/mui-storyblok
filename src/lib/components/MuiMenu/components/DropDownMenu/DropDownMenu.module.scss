.dropdownContent {
  display: none;
  position: absolute;
  min-width: 160px;
  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
  z-index: 99999;
}

.dropdownContent a {
  padding: 12px 16px;
  text-decoration: none;
  display: block;
}

.dropdown:hover .dropdownContent {
  display: block;
}
