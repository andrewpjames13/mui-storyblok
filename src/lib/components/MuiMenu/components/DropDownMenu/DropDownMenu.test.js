import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import DropDownMenu from './DropDownMenu';

function setup() {
  const props = {};
  const comp = shallow(
    <DropDownMenu {...props}>
      <p>here</p>
    </DropDownMenu>,
  );
  return { comp, props };
}

describe('<DropDownMenu />', () => {
  it('renders DropDownMenu', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot with auth', () => {
    const { props } = setup();

    const tree = renderer.create(
      <DropDownMenu {...props}>
        <p>here</p>
      </DropDownMenu>,
    );
    expect(tree).toMatchSnapshot();
  });
});
