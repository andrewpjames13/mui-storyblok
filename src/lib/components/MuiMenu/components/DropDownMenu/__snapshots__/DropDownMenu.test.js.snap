// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDownMenu /> snapshot with auth 1`] = `
<div
  className="dropdown"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      
      <span
        className="MuiButton-endIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
    </span>
  </button>
  <div
    className="MuiPaper-root dropdownContent MuiPaper-outlined MuiPaper-rounded"
  >
    <p>
      here
    </p>
  </div>
</div>
`;
